Framework/Kernel/src/UsageService.cpp

```diff
@@ -51,11 +57,40 @@ bool FeatureUsage::operator<(const FeatureUsage &r) const {
   return false;
 }
 
+/// Convert the stored feature type enum to a string
+std::string FeatureUsage::featureTypeToString() const {
+
+  switch (type) {
+  case FeatureType::Algorithm:
+    return "Algorithm";
+  case FeatureType::Feature:
+    return "Feature";
+  case FeatureType::Interface:
+    return "Interface";
+  }
+  return "Unknown";
+}
+
+/// Convert the stored vector of strings to a string eg Class1->Method1
+std::string FeatureUsage::nameToString() const {
+
+  std::string result = "";
+
+  for (std::string i : name) {
```

Comment:
Initially didn't compile because of trim parameter in for_each. Had to wrap the trim parameter in boost::bind because trim takes an optional locale parameter